server/sonar-server/src/main/java/org/sonar/server/authentication/UserIdentityAuthenticator.java

```diff
@@ -66,6 +66,12 @@ private long register(UserIdentity user, IdentityProvider provider) {
       if (!provider.allowsUsersToSignUp()) {
         throw new NotAllowUserToSignUpException(provider);
       }
+
+      String email = user.getEmail();
+      if (email != null && dbClient.userDao().existsByEmail(dbSession, email)) {
```

Comment:
the English term `existsByEmail` sounds strange. What about `doesEmailExist()` ?